File: 字符串/滑动窗口/1456.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        unordered_set<char> vowels = {'a', 'e', 'i', 'o', 'u'};
        int cnt=0, ans = 0;

        for (int i=0;i<s.size();i++) {
            if (vowels.count(s[i])) cnt++;
            if (i >= k && vowels.count(s[i-k])) cnt--;
            if (i >= k - 1) ans = max(ans, cnt);
        }

        return ans;
    }
};

File: 力扣/字符串/滑动窗口/1456.java
class Solution {
    public int maxVowels(String s, int k) {
        Set<Character> vowels = Set.of('a', 'e', 'i', 'o', 'u');
        int cnt=0, ans = 0;

        for (int i=0;i<s.length();i++) {
            if (vowels.contains(s.charAt(i))) cnt++;
            if (i >= k && vowels.contains(s.charAt(i-k))) cnt--;
            if (i >= k - 1) ans = Math.max(ans, cnt);
        }

        return ans;
    }
}
